native-schema-registry/csharp/AWSGsrSerDe/AWSGsrSerDe.Tests/memory/AwsSchemaRegistryMemeoryLeakDetectionTests.cs

```diff
@@ -0,0 +1,83 @@
+using NUnit.Framework;
+using static AWSGsrSerDe.Tests.utils.ProtobufGenerator;
+
+
+namespace AWSGsrSerDe.Tests.memory
+{
+    [TestFixture]
+    public class AwsSchemaRegistryMemeoryLeakDetectionTests
+    {
+        private static readonly Dictionary<string, dynamic> Configs = new Dictionary<string, dynamic>
+        {
+            { GlueSchemaRegistryConstants.AvroRecordType, AvroRecordType.GenericRecord },
+            { GlueSchemaRegistryConstants.DataFormatType, GlueSchemaRegistryConstants.DataFormat.AVRO },
+        };
+
+        private static readonly GlueSchemaRegistryKafkaSerializer KafkaSerializer = new GlueSchemaRegistryKafkaSerializer(Configs);
+        private static readonly GlueSchemaRegistryKafkaDeserializer KafkaDeserializer = new GlueSchemaRegistryKafkaDeserializer(Configs);
+
+        private void SerializeDeserializeProtobufMessage()
+        {
+            var message = (IMessage)BASIC_SYNTAX2_MESSAGE;
+            var configs = new Dictionary<string, dynamic>
+            {
+                { GlueSchemaRegistryConstants.ProtobufMessageDescriptor, message.Descriptor },
+                { GlueSchemaRegistryConstants.DataFormatType, GlueSchemaRegistryConstants.DataFormat.PROTOBUF },
+            };
+
+            KafkaSerializer.Configure(configs);
+            KafkaDeserializer.Configure(configs);
+
+            var serialized = KafkaSerializer.Serialize(message, message.Descriptor.FullName);
+
+            var deserializedObject =
+                KafkaDeserializer.Deserialize(message.Descriptor.FullName, serialized);
+        }
+
+        private void SerializeDeserializeJsonMessage()
+        {
+            var loadJsonGenericRecord = SchemaLoader.LoadJsonGenericRecord(
+                "schema/draft07/geographical-location.schema.json",
+                "geolocation1.json",
+                true);
+            var message = JsonDataWithSchema.Build(loadJsonGenericRecord.Schema, loadJsonGenericRecord.Payload);
+            var configs = new Dictionary<string, dynamic>
+            {
+                { GlueSchemaRegistryConstants.DataFormatType, GlueSchemaRegistryConstants.DataFormat.JSON },
+            };
+
+            KafkaSerializer.Configure(configs);
+            KafkaDeserializer.Configure(configs);
+
+            var serialized = KafkaSerializer.Serialize(message, "test-topic-json");
+            var deserializedObject = KafkaDeserializer.Deserialize("test-topic-json", serialized);
+        }
+
+        // TODO: Improve and automate the memory leak check
+        // Test for detecting memory leak
+        [Test]
+        public void KafkaSerDeMemoryTestForProtobuf()
+        {
+            for (; ;)
+            {
+                SerializeDeserializeProtobufMessage();
+            }
+        }
+        
+        [Test]
+        public void KafkaSerDeMemoryTestForJson()
+        {
+            for (; ;)
+            {
+                SerializeDeserializeJsonMessage();
+            }
+        }
```

Comment:
Won't only one of them would execute? Shouldn't we combine all of them?

Reply:
Good idea, I will combine them